Vizio: when checking new host against existing config entry hosts, make check hostname aware

raman325 opened the pull request:
## Proposed change
I discovered through discussion of #36265 that if a user uses a hostname instead of an IP address when configuring a config entry, discovery may not realize the host already existed and will notify the user of a new entry. This change makes it so that the host check is hostname aware and resolves hostnames as part of the check. This also impacts the user and import config flows because they will also do the same check, which is safe because it achieves the original intent of the host check. ~I also added a change to ensure that when updating an existing config entry via import, the `data` gets properly updated so that the config entry doesn't continue to get updated or checked for updates on future HA restarts.~


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: #36265
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [x] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

raman325 committed:
make ip check hostname aware

MartinHjelmare commented on homeassistant/components/vizio/config_flow.py:
```diff
@@ -90,7 +92,11 @@ def _get_pairing_schema(input_dict: Dict[str, Any] = None) -> vol.Schema:
 
 def _host_is_same(host1: str, host2: str) -> bool:
```
This now has to run in the executor thread pool as there's I/O inside.

raman325 replied:
Good point. So now this function has to be async I think? If so, how do I handle the path where socket.gethostbyname is never called and there is therefore nothing to block on?

raman325 replied:
Good point. So now this function has to be async I think? If so, how do I handle the path where socket.gethostbyname is never called and there is therefore nothing to block on?

raman325 replied:
Good point. So now this function has to be async I think? If so, how do I handle the path where socket.gethostbyname is never called and there is therefore nothing to block on?

raman325 replied:
‘await asyncio.sleep(0)’ maybe?

MartinHjelmare replied:
I'd schedule this whole function on the executor from the steps where it's needed.

raman325 replied:
better idea 👍

raman325 committed:
add executor job for sync function doing IO and remove errant comma

raman325 committed:
revert change to update new_data explicitly for options keys since it is already done later

raman325 commented:
Just to note, I reverted a change I introduced here that's unrelated to the original intent of the PR (titled) because I realize it's already handled later

raman325 committed:
empty commit to retrigger CI
